File: common/src/main/java/dev/kineticcat/hexportation/api/transfer/StorageView.java
package dev.kineticcat.hexportation.api.transfer;

/**
 * Cross-platform replacement for Fabric's StorageView interface.
 * Provides identical API so existing code needs zero changes.
 */
public interface StorageView<T> {
    
    /**
     * Get the resource stored in this view.
     */
    T getResource();
    
    /**
     * Get the amount stored in this view.
     */
    long getAmount();
    
    /**
     * Get the capacity of this view.
     */
    long getCapacity();
    
    /**
     * Simulate extracting from this view.
     */
    long simulateExtract(T resource, long maxAmount, Object transaction);
    
    /**
     * Extract from this view.
     */
    long extract(T resource, long maxAmount, Object transaction);
}

File: common/src/main/java/dev/kineticcat/hexportation/api/transfer/StorageView.kt
package dev.kineticcat.hexportation.api.transfer

/**
 * Cross-platform replacement for Fabric's StorageView interface.
 * Provides identical API so existing code needs zero changes.
 */
interface StorageView<T> {
    
    /**
     * Get the resource stored in this view.
     */
    val resource: T
    
    /**
     * Get the amount stored in this view.
     */
    val amount: Long
    
    /**
     * Get the capacity of this view.
     */
    val capacity: Long
    
    /**
     * Check if this view is blank (empty).
     */
    val isBlank: Boolean
    
    /**
     * Simulate extracting from this view.
     */
    fun simulateExtract(resource: T, maxAmount: Long, transaction: Any?): Long
    
    /**
     * Extract from this view.
     */
    fun extract(resource: T, maxAmount: Long, transaction: Any?): Long
}